viewer/packages/rendering/src/cameraconfig.ts

```diff
@@ -13,7 +13,26 @@ export interface SuggestedCameraConfig {
   far: number;
 }
 
-export function suggestCameraConfig(rootSector: SectorMetadata): SuggestedCameraConfig {
+const MAX_VERTICAL_ANGLE = (30 / 180) * Math.PI;
+
+function ensureYAngleBelowThresholdForNormalizedVector(direction: THREE.Vector3): void {
+  if (direction.y >= Math.sin(MAX_VERTICAL_ANGLE)) {
+    const xzProjection = new THREE.Vector2(direction.x, direction.z);
+    const xzLength = xzProjection.length();
+    if (xzLength <= 1e-4) {
+      xzProjection.x = 1.0;
+      xzProjection.y = 1.0;
+    }
+
+    xzProjection.normalize().multiplyScalar(Math.cos(MAX_VERTICAL_ANGLE));
+    direction.x = xzProjection.x;
+    direction.z = xzProjection.y;
+
+    direction.y = Math.sin(MAX_VERTICAL_ANGLE);
+  }
```

Comment:
Concider inverting the if statement to reduce nesting. Also, the `xzProjection` can be covered using the ternary operator.